### PR TITLE
Link-preview and tab saving

### PR DESCRIPTION
**Pull request by ZsokaM**
Users can now save links for later, once saved there will be a short link preview displayed. 
Adjusted some styling.
Updated models and adjusted delete, newtab routes.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ app.use(
     cookie: {
       sameSite: true,
       httpOnly: true,
-      maxAge: 60000,
+      maxAge: 120000,
```
Comment (michaelkoek) [inline app.js]: Maybe best to set it for a whole year? `365 * 24 * 60 * 60 * 1000`
Or did you set it to this deliberately ?

File: models/Tabs.model.js [hunk 2]
```diff
@@ -2,22 +2,11 @@ const { Schema, model } = require("mongoose");
...
-      required: [true, "Select a category: video, article, podcast"],
-      trim: true,
-    },
-    title: {
-      type: String,
-      required: [true, "Add a short title"],
-      trim: true,
-    },
+    category: String,
+    title: String,
```
Comment (michaelkoek) [inline models/Tabs.model.js:6]: A few of these I would assume would be required? Or why not?

File: public/stylesheets/style.css [hunk 3]
```diff
@@ -72,6 +77,9 @@ main {
 
 .logo {
   width: 50px;
+  text-decoration: none;
+  color: white;
+  font-size: 2em;
```
Comment (michaelkoek) [inline public/stylesheets/style.css:82]: Why not `rem`?

File: public/stylesheets/style.css [hunk 4]
```diff
@@ -220,6 +234,36 @@ main {
...
+  display: flex;
+  background-color: transparent;
+  border: 1px solid white;
+  width: 75%;
+  color: white;
+  font-size: 0.75em;
+  margin: 0.5em;
+  align-items: center;
+  line-height: 1;
+  padding: 0.25em;
```
Comment (michaelkoek) [inline public/stylesheets/style.css:259]: In this whole class "Why not `rem`?

File: routes/user.routes.js [hunk 5]
```diff
@@ -73,4 +82,43 @@ router.post("/:folderId/edit", (req, res, next) => {
     .catch((err) => next(err));
 });
 
+router.post("/newtab", async (req, res, next) => {
+  const { category, description } = req.body;
+
+  const link = await axios.post("https://api.linkpreview.net", {
+    q: `${description}`,
```
Comment (michaelkoek) [inline routes/user.routes.js]: No need to use string interpellation as this is already a javascript variable. Only use it when you are combining it with text

File: routes/user.routes.js [hunk 6]
```diff
@@ -73,4 +82,43 @@ router.post("/:folderId/edit", (req, res, next) => {
...
+    key: `${process.env.LINK_KEY}`,
```
Comment (michaelkoek) [inline routes/user.routes.js]: No need to use string interpellation as this is already a javascript variable. Only use it when you are combining it with text

File: routes/user.routes.js [hunk 7]
```diff
@@ -73,4 +82,43 @@ router.post("/:folderId/edit", (req, res, next) => {
...
+  });
+
+  Tab.create({
+    category,
+    title: link.data.title,
```
Comment (michaelkoek) [inline routes/user.routes.js]: A perfect use case to use Destructering

File: views/users/profile.hbs [hunk 8]
```diff
@@ -25,27 +25,52 @@
   </section>
 
   <section class="side side-right">
-    <a href="#"><img src="Images/add.png" width="50px" disabled/></a>
+    <form action="/newtab" method="POST" >
+      <div class="form-div">
+        <select name="category" class="form-input" height="10">
+          <option value=" ">Pick a category</option>
```
Comment (michaelkoek) [inline views/users/profile.hbs]: Remove the empty white space. no need for it

File: views/users/profile.hbs [hunk 9]
```diff
@@ -25,27 +25,52 @@
...
+          <option name="video" value="video">Video</option>
+          <option name="article" value="article">Article</option>
+          <option name="podcast" value="podcast">Podcast or Audio</option>
+          <option name="book" value="book">Book</option>
+          <option name="fullsite" value="fullsite">Web site</option>
+        </select>
+      </div>
+      <div class="form-div">
+        <input type="text" name="description" class="form-input" placeholder="Add the link"></input>
```
Comment (michaelkoek) [inline views/users/profile.hbs]: `<input />` is a self closing element so remove the `</input>` and add the slash at the opening one.
Second; is this a one line element? if yes, than it's okay, but as the name is `description` I would assume multiline which would make it a `<textarea></textarea>`.

So change the name if it's a one line

File: views/users/profile.hbs [hunk 10]
```diff
@@ -25,27 +25,52 @@
...
+              <a href="{{this.url}}">{{this.title}}</a>
+              <p>{{this.description}}</p>
+              <form action="/newtab/{{this._id}}/delete" method="POST">
+                <button type="submit" class="btn" style="font-size: smaller;">Delete</button>
+              </form>
+            </div>
+          </article>
+        {{/each}}
+      {{else}}
+        <h3 class="text-center" style="color:white">Save nice videos, articles and podcasts for later</h3>
```
Comment (michaelkoek) [inline views/users/profile.hbs]: Why do you add inline styling here? Especially when you already have a `class` defined.
Don't use inline styling

File: models/Tabs.model.js (hunk 2 above)
Comment (ZsokaM) [reply models/Tabs.model.js:6]: hmm the category could be required, but as the rest is coming from the LinkPreview API and not all links will have images etc I didnt want to make it compulsory.

File: app.js (hunk 1 above)
Comment (ZsokaM) [reply app.js]: no deliberate reason, 30 min sounded reasonable but can increase.

File: public/stylesheets/style.css (hunk 3 above)
Comment (ZsokaM) [reply public/stylesheets/style.css:82]: im adding this to the Michael app :) ill update to rem where the root size makes more sense.

File: routes/user.routes.js (hunk 5 above)
Comment (hisamparker) [reply routes/user.routes.js]: Unless you want to make sure the value is type string?

**Comment by hisamparker** [comment]
I think you're cool, you picked a really difficult project. Very rad 👯‍♀️

File: routes/user.routes.js (hunk 5 above)
Comment (michaelkoek) [reply routes/user.routes.js]: `req.body.description` already comes as a string value

File: views/users/profile.hbs [hunk 11]
```diff
@@ -25,27 +25,52 @@
...
-    <a href="/profile" class="btn">Back</a>
+    <section class="tabs">
+      {{#if user.tabs}}
+        {{#each user.tabs}}
+          <article class="link-container">
+            <div>
+              {{#if this.image}}
+                <img src="{{this.image}}" alt="link-image" class="link-img"></img>
+              {{else}}
+                <img src="Images/link.png" alt="link-default-image" class="link-img"></img>
```
Comment (michaelkoek) [inline views/users/profile.hbs]: the `` tag is a self-closing tag. This comment also applies to line 51

File: views/users/profile.hbs [hunk 12]
```diff
@@ -25,27 +25,52 @@
...
+              {{/if}}
+            </div>
+            <div>
+              <h3>{{this.category}}</h3>
+              <a href="{{this.url}}">{{this.title}}</a>
+              <p>{{this.description}}</p>
+              <form action="/newtab/{{this._id}}/delete" method="POST">
+                <button type="submit" class="btn" style="font-size: smaller;">Delete</button>
```
Comment (michaelkoek) [inline views/users/profile.hbs:61]: Don't use inline styling

File: routes/user.routes.js [hunk 13]
```diff
@@ -73,4 +82,50 @@ router.post("/:folderId/edit", (req, res, next) => {
...
+    Tab.create({
+      category,
+      title,
+      description: shortDescript,
+      image,
+      url,
+      user: req.user,
+    })
+      .then((tab) => {
+        console.log(tab);
```
Comment (michaelkoek) [inline routes/user.routes.js]: Remove the `console.log()`, this should also be used for development purposes

File: public/stylesheets/style.css [hunk 14]
```diff
@@ -220,6 +234,39 @@ main {
...
+  flex-direction: column;
+  justify-content: center;
+  align-content: center;
+  align-items: center;
+}
+.tab h3 {
+  color: white;
+}
+
+.link-container {
```
Comment (michaelkoek) [inline public/stylesheets/style.css]: nit pick:
When I see 'link' I assume it's about an `<a>` (anchor). So maybe you could give a bit more descriptive name like `link-preview-[other things]`

**Pull request merged**